packages/alfa-dom/src/h.ts

```diff
@@ -0,0 +1,291 @@
+      type: "fragment",
+      children: children.map((child) =>
+        typeof child === "string" ? text(child) : child
+      ),
+    };
+  }
+
+  export function sheet(
+    rules: Array<Rule.JSON>,
+    disabled: boolean = false,
+    condition: string | null = null
+  ): Sheet.JSON {
+    return {
+      rules,
+      disabled,
+      condition,
+    };
+  }
+
+  export function block(
+    declarations: Array<Declaration.JSON> | Record<string, string>
+  ): Block.JSON {
+    return Array.isArray(declarations)
+      ? declarations
+      : entries(declarations).map(([name, value]) => {
+          const important = value.endsWith("!important");
+
+          value = value.replace(/\s+?!important$/, "");
+
+          return {
+            name: hyphenate(name),
+            value,
+            important,
+          };
+        });
+  }
+
+  export function declaration(
+    name: string,
+    value: string,
+    important: boolean = false
+  ): Declaration.JSON {
+    return {
+      name,
+      value,
+      important,
+    };
+  }
+
+  export namespace rule {
+    export function fontFace(
+      declarations: Array<Declaration.JSON> | Record<string, string>
+    ): FontFaceRule.JSON {
+      return {
+        type: "font-face",
+        style: block(declarations),
+      };
+    }
+
+    export function keyframe(
+      key: string,
+      declarations: Array<Declaration.JSON> | Record<string, string>
+    ): KeyframeRule.JSON {
+      return {
+        type: "keyframe",
+        key,
+        style: block(declarations),
+      };
+    }
+
+    export function keyframes(
+      name: string,
+      rules: Array<Rule.JSON>
+    ): KeyframesRule.JSON {
+      return {
+        type: "keyframes",
+        name,
+        rules,
+      };
+    }
+
+    export function media(
+      condition: string,
+      rules: Array<Rule.JSON>
+    ): MediaRule.JSON {
+      return {
+        type: "media",
+        condition,
+        rules,
+      };
+    }
+
+    export function namespace(
+      namespace: string,
+      prefix: string | null = null
+    ): NamespaceRule.JSON {
+      return {
+        type: "namespace",
+        namespace,
+        prefix,
+      };
+    }
+
+    export function page(
+      selector: string,
+      declarations: Array<Declaration.JSON> | Record<string, string>
+    ): PageRule.JSON {
+      return {
+        type: "page",
+        selector,
+        style: block(declarations),
+      };
+    }
+
+    export function style(
+      selector: string,
+      declarations: Array<Declaration.JSON> | Record<string, string>
+    ): StyleRule.JSON {
+      return {
+        type: "style",
+        selector,
+        style: block(declarations),
+      };
+    }
+
+    export function supports(
+      condition: string,
+      rules: Array<Rule.JSON>
+    ): SupportsRule.JSON {
+      return {
+        type: "supports",
+        condition,
+        rules,
+      };
+    }
+  }
+}
+
+function hyphenate(value: string): string {
```

Comment:
Should that be called `camelCaseToKebabCase`?

Reply:
That's a terribly long name for a one-line module-internal function 🙈 I would have inlined it but needed to call it twice.